rusoto/core/src/client.rs

```diff
@@ -86,6 +90,13 @@ impl Client {
     ) -> RusotoFuture<T, E> {
         future::new(self.inner.sign_and_dispatch(request), response_handler)
     }
+
+    /// Set content_encoding type for http requests which will compress request payload accordingly
+    pub fn set_http_content_encoding(&mut self, content_encoding: ContentEncoding) {
+        let inner = Arc::get_mut(&mut self.inner)
```

Comment:
this is the main problem I see - sometimes it will succeed, sometimes it will panic, depending on the number of references.